### ISSUE TITLE
Anki crashes when deleting deck and having Browser open at the same time

### ISSUE THREAD

**Issue by Mauville** [ankitects/anki#1107]
When deleting a deck from the Decks meny and having the Browser open in that deck at the same time, Anki outputs the attached error. After this, Anki will refuse to close unless killed in Task Manager.

I've disabled addons and successfully replicated the error with a new deck with one Basic card. If the deck is not selected in the browser, the operation completes as it normally would.

```
Error
An error occurred. Please use Tools > Check Database to see if that fixes the problem.
If problems persist, please report the problem on our support site. Please copy and paste the information below into your report.
Anki 2.1.40 (cf446733) Python 3.8.6 Qt 5.14.2 PyQt 5.14.2
Platform: Windows 10
Flags: frz=True ao=False sv=1
Add-ons, last update check: 2021-03-30 13:45:03

Caught exception:
Traceback (most recent call last):
  File "aqt\browser.py", line 708, in closeEvent
  File "aqt\editor.py", line 508, in saveNow
  File "aqt\editor.py", line 617, in saveTags
  File "anki\notes.py", line 68, in flush
  File "anki\rsbackend_gen.py", line 364, in update_note
  File "anki\rsbackend.py", line 256, in _run_command
anki.rsbackend.NotFoundError


```

**Comment by BlueGreenMagick** [comment]
Can confirm that this issue is in 2.1.42, but seems to have been fixed in the latest master. 

In aqt/table.py
```python
    def _fetch_row_from_backend(self, item: ItemId) -> CellRow:
        try:
            row = CellRow(*self.col.browser_row_for_id(item))
        except NotFoundError: # <- This seems to have fixed the issue 
            return CellRow.deleted(self.len_columns())
        except Exception as e:
            return CellRow.generic(self.len_columns(), str(e))


```

**Comment by dae** [comment]
That's unrelated, this was actually fixed in 8de6113e8a3809e47799e15ba96ac7006eea2f44

**Comment by dae** [comment]
(and it should be in the 2.1.43 release from today)
